### ISSUE TITLE
Build-Variables are treated as properties of the task group

### ISSUE THREAD

**Issue by aKzenT** [rfennell/vNextBuild#103]
### VSTS Extension you are using
GenerateReleaseNotes 

### Where are you running it?
VSTS

### Version of Extension/Task
Latest Version

### Expected behavior and actual behavior
I'm using placeholders like $($build.buildnumber) as an inline template. Since the task is part of a task group, VSTS treats this as a variable and exposes it as a property to be set on the task group. The expected behaviour is to be able to use the placeholders without VSTS intefering.

### Steps to reproduce the problem
- Create a new task group with a GenerateReleaseNotes task.
- Use an inline template and enter placeholders like $($build.buildnumber).
- Switch to the properties tab of the task group.
- VSTS created a new property entry with the name of "$build.buildnumber". All builds or releases using this task group have to specify a value for this property.

**Comment by rfennell** [comment]
I need to think about this one, and I fear that it is going to be beyond my fixing being down to the way tasks groups are implemented.

That said, I am surprised, all the task does is get build/release environment variable then call the API. I need to see why the task group is interepting these values.

**Comment by rfennell** [comment]
I have had a more detailed look at this one, as I suspected this is an issue with the way the VSTS UI processes the task properties when it creates a task group. It sees all variables in an inline script as things to be expanded - which is wrong but beyond my control.

However, there is a work around. Don't use an inline template. I have successfully been able to create a task group that contains a generate release notes task without any problems.
